libqtile/lazy.py

```diff
@@ -144,6 +163,16 @@ def check(self, q) -> bool:
         if self._layouts and q.current_layout.name not in self._layouts:
             return False
 
+        if self._func:
+            try:
+                result = self._func()
+            except Exception:
+                logger.exception("Error when running function in lazy call. Ignoring.")
+                result = True
+
+            if not result:
+                return False
+
         return True
```

Comment:
With the typing change above to `self._func`, this whole bit could just be:
```suggestion
        return self._func()
```
Assuming of course that people's configs are type-safe, which would give us some confidence that an error won't be raised. Ha. Yeah, good to catch an exception :)

Reply:
As useful as typing is, I don't see it as a replacement for error handling!